analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/DefaultValue.cs

```diff
@@ -0,0 +1,32 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class DefaultValue : ISimpleProcessor
+{
+    public ProgramState Process(SymbolicContext context) =>
+        context.Operation.Instance.Type.OriginalDefinition.Is(KnownType.System_Nullable_T)
```

Comment:
This should use the extension method added in the other PR.

Reply:
I like the idea and think that this is how we should fix the nullable support. (For unconstraint generics, we may branch here).

Reply:
Shouldn't we set the "Null" constraint for reference types also?

Reply:
Reference types are done in `ConstantCheck`, because they have `.ConstantValue`